ocs_sample_library_preview/Securable.py

```diff
@@ -0,0 +1,210 @@
+        return result
+
+    def updateDefaultAccessControl(self, namespace_id: str, access_control: AccessControlList):
+        """
+        Update a collection's default access control list
+        :param access_control: The access control list
+        """
+        if access_control is None:
+            raise TypeError
+
+        response = self.__base_client.request(
+            'put', self.__default_collection_acl_path(
+                tenant_id=self.__tenant,
+                namespace_id=namespace_id),
+            data=access_control.toJson())
+        self.__base_client.checkResponse(
+            response, f'Failed to update collection access control, {self.__collection}.')
+
+    def patchDefaultAccessControl(self, namespace_id: str, patch: JsonPatch):
+        """
+        Patch a collection's default access control list
+        :param patch: A JSON Patch document describing the patch operations
+        """
+        if patch is None:
+            raise TypeError
+
+        response = self.__base_client.request(
+            'patch', self.__default_collection_acl_path(
+                tenant_id=self.__tenant,
+                namespace_id=namespace_id),
+            data=patch.to_string())
+        self.__base_client.checkResponse(
+            response, f'Failed to patch collection access control, {self.__collection}.')
+
+    def getAccessControl(self, namespace_id: str, item_id: str) -> AccessControlList:
+        """
+        Get an item's access control list
+        :param item_id: The item identifier
+        """
+        if item_id is None:
+            raise TypeError
+
+        response = self.__base_client.request(
+            'get', self.__item_acl_path.format(
+                tenant_id=self.__tenant,
+                namespace_id=namespace_id,
+                item_id=item_id))
+        self.__base_client.checkResponse(
+            response, f'Failed to get access control, {item_id} in collection {self.__collection}.')
+
+        result = AccessControlList.fromJson(response.json())
+        return result
+
+    def updateAccessControl(self, namespace_id: str, item_id: str, access_control: AccessControlList):
+        """
+        Update an item's access control list
+        :param item_id: The item identifier
+        :param access_control: The access control list
+        """
+        if item_id is None:
+            raise TypeError
+        if access_control is None:
+            raise TypeError
+
+        response = self.__base_client.request(
+            'put', self.__item_acl_path.format(
+                tenant_id=self.__tenant,
+                namespace_id=namespace_id,
+                item_id=item_id),
+            data=access_control.toJson())
+        self.__base_client.checkResponse(
+            response, f'Failed to update access control, {item_id} in collection {self.__collection}.')
+
+    def patchAccessControl(self, namespace_id: str, item_id: str, patch: JsonPatch):
```

Comment:
Do we care that not every endpoint supports `PATCH`? I think the only way to handle that distinction would be to have a second class that inherits Securable and extends it with the collection and item patch operations, then use the appropriate Securable type on each endpoint. I don't know whether there's a plan to implement this on every securable endpoint eventually.

Reply:
I didn't realize that they didn't all support Patch. I imagine that you would get a 404 error back if it doesn't exist, so maybe the response can be checked to give a better error message that mentions that the endpoint may not support ```PATCH```. Your idea would also work so I'm not sure what would be best. Any thoughts from @kelseyduffy or @derekendres?

Reply:
my thoughts are that doug's idea is the most elegant/correct. if we plan to implement it eventually for all of them, though, then we could probably get away with something more temporary such as an array of 'currently not patchable' endpoints or something. then if you get a 404, check if the endpoint is in that list; if so, say it's not supported; if not, say it's an error. it's ugly but minimal effort. if we don't plan to implement it on all the securable classes, though, then it would be worth the effort to create a patchable extension.

Reply:
I think you are both right that it should be a separate class. I implemented a ```PatchableSecurable``` class for ```Streams```, ```Types```, and ```StreamViews``` to inherit.